### PR TITLE
upgrade go version to 1.15

### PR DESCRIPTION
**Pull request by marcolan018**
Signed-off-by: Marco <llan@redhat.com>

### CODE REVIEW

File: Dockerfile.prow [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Copyright Contributors to the Open Cluster Management project
 
-FROM openshift/origin-release:golang-1.13 AS builder
+FROM openshift/origin-release:golang-1.15 AS builder
```
Comment (morvencao) [inline Dockerfile.prow]: I found other observability repos are using `registry.ci.openshift.org/open-cluster-management/builder:go1.15-linux` as builder image, is there any special reason we use this for metric collector repo?

Comment (marcolan018) [reply Dockerfile.prow]: @morvencao 
we inherit to use this image from telmetry client. Do you think we need to change to use the one in registry.ci.openshift-org?

Comment (morvencao) [reply Dockerfile.prow]: I found we have add builder image `registry.ci.openshift.org/open-cluster-management/builder:go1.15-linux-amd64` in prow config for metric-collector
https://github.com/openshift/release/blob/459db96c7779c2eb81713eadb1f732686c6102d8/ci-operator/config/open-cluster-management/metrics-collector/open-cluster-management-metrics-collector-main.yaml#L6-L10

And I'm updating it to multiarch image in https://github.com/openshift/release/pull/17696

Comment (clyang82) [reply Dockerfile.prow]: we may use `open-cluster-management` image as suggested by cicd.

Comment (marcolan018) [reply Dockerfile.prow]: @morvencao @clyang82 
I updated the builder image

**Review by morvencao — COMMENTED** [review]
/lgtm

**Pull request merged**